Content.Server/StationEvents/Events/RandomSentienceRule.cs

```diff
@@ -45,24 +67,15 @@ protected override void Started(EntityUid uid, RandomSentienceRuleComponent comp
         var kind2 = groupList.Count > 1 ? groupList[1] : "???";
         var kind3 = groupList.Count > 2 ? groupList[2] : "???";
 
-        foreach (var target in targetList)
-        {
-            var station = StationSystem.GetOwningStation(target);
-            if(station == null)
-                continue;
-            stationsToNotify.Add((EntityUid) station);
-        }
-        foreach (var station in stationsToNotify)
-        {
-            ChatSystem.DispatchStationAnnouncement(
-                station,
-                Loc.GetString("station-event-random-sentience-announcement",
-                    ("kind1", kind1), ("kind2", kind2), ("kind3", kind3), ("amount", groupList.Count),
-                    ("data", Loc.GetString($"random-sentience-event-data-{RobustRandom.Next(1, 6)}")),
-                    ("strength", Loc.GetString($"random-sentience-event-strength-{RobustRandom.Next(1, 8)}"))),
-                playDefaultSound: false,
-                colorOverride: Color.Gold
-            );
-        }
+        ChatSystem.DispatchStationAnnouncement(
+            station.Value,
+            Loc.GetString("station-event-random-sentience-announcement",
+                ("kind1", kind1), ("kind2", kind2), ("kind3", kind3), ("amount", groupList.Count),
+                ("data", Loc.GetString(_random.Pick(_prototype.Index<LocalizedDatasetPrototype>("RandomSentienceEventData").Values))),
+                ("strength", Loc.GetString(_random.Pick(_prototype.Index<LocalizedDatasetPrototype>("RandomSentienceEventStrength").Values)))
```

Comment:
```suggestion
                ("data", _random.Pick(_prototype.Index<LocalizedDatasetPrototype>("RandomSentienceEventData"))),
                ("strength", _random.Pick(_prototype.Index<LocalizedDatasetPrototype>("RandomSentienceEventStrength")))
```
There's a helper that handles the localization automatically. You'll need to add `using Content.Shared.Random.Helpers;` too.